### ISSUE TITLE
UPGD adaptive_kappa_mode silently discards the configured bounder (its kappa and its bounding law), even during warm-up

### ISSUE THREAD

**Issue by Prabhat1308** [SlopDotCash/asi#392]
## Summary

`UPGDLearner` (`alberta_framework/core/upgd.py`) documents `adaptive_kappa_mode` as "Optional online control law for ObGD `kappa`. `"none"` uses the configured bounder unchanged." Under `"loss_ratio"` / `"gradient_alignment"` the update path calls `_obgd_bound_with_kappa(...)` with `effective_kappa` derived from `adaptive_kappa_base` and never consults `self._bounder` — including inside `adaptive_kappa_warmup_steps`, where it "falls back" to `adaptive_kappa_base`, not to the configured bounder's kappa. So a configured bounder is silently replaced:

- `ObGDBounding(kappa=2.0)` + adaptive mode (base 0.5): the effective step changes 4× with the adaptive law provably inert (`warmup_steps=1_000_000`);
- `AdaptiveObGDBounding(kappa=1e-4)` + adaptive mode with `min=max=base=1e-4` (kappa bit-identical): the whole RMS stage vanishes and the update collapses to plain ObGD (`-0.0063 → -1.7003`, 268×).

Any ablation that flips `adaptive_kappa_mode` on/off with such a bounder measures a bounder swap and attributes it to the adaptive law.

## Falsifier (deterministic, pristine `main` @ `4f75cc8`)

```python
import jax.numpy as jnp, jax.random as jr
from alberta_framework.core.optimizers import AdaptiveObGDBounding, ObGDBounding
from alberta_framework.core.upgd import UPGDLearner
def delta(bounder, mode):
    L = UPGDLearner(n_heads=1, hidden_sizes=(), sparsity=0.0, step_size=0.5, perturbation_sigma=0.0,
                    bounder=bounder, adaptive_kappa_mode=mode, adaptive_kappa_base=1e-4,
                    adaptive_kappa_min=1e-4, adaptive_kappa_max=1e-4, adaptive_kappa_warmup_steps=1_000_000)
    s = L.init(feature_dim=3, key=jr.key(0)); w0 = s.head_params.weights[0]
    r = L.update(s, jnp.array([1., -0.5, 0.25]), jnp.array([3.0]))
    return float(jnp.sum(r.state.head_params.weights[0] - w0))
print("AdaptiveObGD kappa_mode='none'      ", delta(AdaptiveObGDBounding(kappa=1e-4), "none"))
print("AdaptiveObGD kappa_mode='loss_ratio'", delta(AdaptiveObGDBounding(kappa=1e-4), "loss_ratio"))
print("plain ObGD   kappa_mode='none'      ", delta(ObGDBounding(kappa=1e-4), "none"))
```

Output on `main`:

```
AdaptiveObGD kappa_mode='none'       -0.0063417
AdaptiveObGD kappa_mode='loss_ratio' -1.7003381
plain ObGD   kappa_mode='none'       -1.7003381
```

The adaptive branch is inert here (warm-up never ends, kappa pinned to 1e-4 on both sides); the entire difference is the configured `AdaptiveObGDBounding` being replaced by plain ObGD.

Expected: since the adaptive modes bound with plain ObGD at `adaptive_kappa_base` by design (the shipped `step2_strict_digit_readout_default` pairs `ObGDBounding(kappa=0.5)` with `adaptive_kappa_base=0.5`, and one in-repo test uses `bounder=None`), a configured bounder that the adaptive branch would silently discard — any non-`ObGDBounding`, or an `ObGDBounding` whose kappa differs from `adaptive_kappa_base` — is rejected at construction, and the contract is stated in the docstring.

## Dedup

`gh issue/pr list --search "adaptive_kappa" / "bounder"`: nothing on this surface (PRs #61/#69 are the ObGD 0×inf lane). Related open (mine, different code path in the same module): #374/#375; merge-tree composes cleanly.

Fix PR follows (failing-test-first).
